### PR TITLE
SA1207 analyzer added.

### PR DESCRIPTION
**Pull request by EdwinEngelen**
My first analyzer implementation. Any tips? :smile:

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1207ProtectedMustComeBeforeInternal.cs [hunk 1]
```diff
@@ -46,7 +49,44 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
...
+                SyntaxKind.InterfaceDeclaration,
+                SyntaxKind.MethodDeclaration,
+                SyntaxKind.PropertyDeclaration,
+                SyntaxKind.StructDeclaration);
+        }
+
+        private static void HandleDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            // TODO: Should we do null checking? It's not covered in a test, because (this version of) Roslyn never returns null.
+            var childTokens = context.Node?.ChildTokens()?.ToArray();
```
Comment (Maxwe11) [inline StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1207ProtectedMustComeBeforeInternal.cs]: remove `ToArray()`

Comment (EdwinEngelen) [inline StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1207ProtectedMustComeBeforeInternal.cs]: That's no longer required indeed. Sorry.

File: StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1207CodeFixProvider.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            {
+                context.RegisterCodeFix(CodeAction.Create(OrderingResources.SA1207CodeFix, token => GetTransformedDocumentAsync(context.Document, diagnostic, token)), diagnostic);
+            }
+
+            return SpecializedTasks.CompletedTask;
+        }
+
+        private static async Task<Document> GetTransformedDocumentAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            ////var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```
Comment (Maxwe11) [inline StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1207CodeFixProvider.cs]: Why this commented out? Implement it or not include in this PR

Comment (EdwinEngelen) [inline StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1207CodeFixProvider.cs]: I just wanted a review on the analyzer. Not on the fix. It's not a 'true pull request, but I've been requested to create one.

Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1207CodeFixProvider.cs]: :bulb: It's fine to create the pull request before it's complete (I can label it as **in progress**). However, it helps if you explicitly mention this in the comment for the pull request. As-is, you asked for *any* tips and of course this one would be a tip. :smile:

**Comment by sharwell** [comment]
@EdwinEngelen There's probably no need to close this, unless you are planning to send it with a different branch. Just make your revisions and push them to the same branch, and the pull request will be automatically updated to include them. :smile:

**Comment by EdwinEngelen** [comment]
@sharwell I figured out how to make my branch visible to the public. I created a new pull request. See https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/1011.

**Comment by sharwell** [comment]
Marked as **duplicate** since it's been superseded.

**Pull request closed without merging**